frontend/src/tests/applicant-tests.js

```diff
@@ -126,7 +128,101 @@ export function applicantTests(api) {
-    it.todo("delete an applicant with no associated assignments");
+    it("delete an applicant with no associated assignments", async () => {
+        // upsert and delete an applicant without a session
+        const newApplicant1 = {
+            first_name: "Tommy3",
+            last_name: "Smith3",
+            utorid: "smith3",
+        };
+        const resp = await apiPOST(`/admin/applicants`, newApplicant1);
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(applicantPropTypes, resp.payload);
+        Object.assign(newApplicant1, resp.payload);
+
+        // try deleting the applicant
+        const resp2 = await apiPOST(`/admin/applicants/delete`, {
+            id: newApplicant1.id,
+        });
+        expect(resp2).toHaveStatus("success");
+
+        // verify the applicant is deleted
+        const resp3 = await apiGET(`/admin/applicants`);
+        expect(resp3).toHaveStatus("success");
+        expect(resp3.payload.map((x) => x.id)).not.toContain(newApplicant1.id);
+
+        // upsert and delete an applicant with a session
+        const newApplicant2 = {
+            first_name: "Tommy4",
+            last_name: "Smith4",
+            utorid: "smith4",
+        };
+        const resp4 = await apiPOST(
+            `/admin/sessions/${session.id}/applicants`,
+            newApplicant2
+        );
+        expect(resp4).toHaveStatus("success");
+        checkPropTypes(applicantPropTypes, resp4.payload);
+        Object.assign(newApplicant2, resp4.payload);
+
+        // try deleting the applicant
+        const resp5 = await apiPOST(`/admin/applicants/delete`, {
+            id: newApplicant2.id,
+        });
+        expect(resp5).toHaveStatus("success");
+
+        // verify the applicant is deleted
+        const resp6 = await apiGET(`/admin/sessions/${session.id}/applicants`);
+        expect(resp6).toHaveStatus("success");
+        expect(resp6.payload.map((x) => x.id)).not.toContain(newApplicant2.id);
+    });
+
+    it("fail to delete an applicant with an associated assignment", async () => {
+        // create and insert a new applicant
+        const newApplicant1 = {
+            first_name: "Tommy4",
+            last_name: "Smith4",
+            utorid: "smith4",
+        };
+        const resp = await apiPOST(`/admin/applicants`, newApplicant1);
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(applicantPropTypes, resp.payload);
+        Object.assign(newApplicant1, resp.payload);
+
+        // create and insert a new position
+        const newPosition = {
+            position_code: "CSC100F",
+            position_title: "Basic Computer Science",
+            hours_per_assignment: 70,
+            start_date: "2019/09/09",
+            end_date: "2019/12/31",
+            contract_template_id: contractTemplate.id,
+        };
+
+        const { payload: position } = await apiPOST(
+            `/admin/sessions/${session.id}/positions`,
+            newPosition
+        );
```

Comment:
In order to ensure an assignment is successfully posted, I believe a check here to see if the position is successfully posted is needed.